File: ai/entity/aplist/projectEntity.py
from sqlalchemy import Column, Integer, String
from sqlalchemy.orm import relationship
from sqlalchemy.ext.declarative import declarative_base

Base = declarative_base()

class ProjectEntity(Base):
    __tablename__ = "project"

    key = Column(Integer, primary_key=True, autoincrement=True)  # 프로젝트 고유 키
    id = Column(String(20), unique=True, nullable=False)  # 프로젝트 아이디

    # 관계 설정
    apiLists = relationship("ApiList", back_populates="project", cascade="all, delete-orphan")
    requests = relationship("Request", back_populates="project", cascade="all, delete-orphan")
    filePaths = relationship("FilePath", back_populates="project", cascade="all, delete-orphan")
    dbDesigns = relationship("DbDesign", back_populates="project", cascade="all, delete-orphan")

    def __init__(self, id: str):
        self.id = id

File: backend/spec/src/main/java/com/sk/skala/axcalibur/spec/global/entity/ProjectEntity.java
package com.sk.skala.axcalibur.spec.global.entity;

import java.util.ArrayList;
import java.util.List;

import com.sk.skala.axcalibur.spec.feature.project.entity.ApiListEntity;
import com.sk.skala.axcalibur.spec.feature.project.entity.DbDesignEntity;
import com.sk.skala.axcalibur.spec.feature.project.entity.RequestEntity;
import com.sk.skala.axcalibur.spec.feature.report.entity.ScenarioEntity;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Getter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "project")
@ToString
public class ProjectEntity extends BaseTimeEntity {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  @Column(name = "`key`")
  private Integer key; // 프로젝트 키 (PK, AUTO_INCREMENT)

  @Column(name = "id", unique = true, nullable = false, length = 20)
  private String id; // 프로젝트 ID (UNIQUE)

  @OneToMany(mappedBy = "projectKey", cascade = CascadeType.REMOVE, orphanRemoval = true)
  @Builder.Default
  private List<ApiListEntity> apiLists = new ArrayList<>();

  @OneToMany(mappedBy = "projectKey", cascade = CascadeType.REMOVE, orphanRemoval = true)
  @Builder.Default
  private List<RequestEntity> requests = new ArrayList<>();

  @OneToMany(mappedBy = "projectKey", cascade = CascadeType.REMOVE, orphanRemoval = true)
  @Builder.Default
  private List<DbDesignEntity> dbDesigns = new ArrayList<>();

  @OneToMany(mappedBy = "projectKey", cascade = CascadeType.REMOVE, orphanRemoval = true)
  @Builder.Default
  private List<ScenarioEntity> scenarios = new ArrayList<>();

  @OneToMany(mappedBy = "projectKey", cascade = CascadeType.REMOVE, orphanRemoval = true)
  @Builder.Default
  private List<FilePathEntity> filePaths = new ArrayList<>();

  public ProjectEntity(String id) {
    this.id = id;
  }
}